sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -0,0 +1,427 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+
+#nullable enable
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// A policy that sends an <see cref="AccessToken"/> provided by a <see cref="TokenCredential"/> as an Authentication header.
+    /// </summary>
+    internal class BearerTokenChallengeAuthenticationPolicy : HttpPipelinePolicy
+    {
+        private const string AuthenticationChallengePattern = @"(\w+) ((?:\w+="".*?""(?:, )?)+)(?:, )?";
```

Comment:
The `.*?` doesn't make sense. I'm surprised it's even valid, but `*` would mean 0 or more while `?` means 0 or 1. Which is it? Also, while not likely, you might allow for `(?:,\s+)`.

Reply:
It does, the `?` means as few as possible in this case. https://regex101.com/r/z0ZAqD/1

Reply:
Interesting. I've been a long-time regex (ab)user and never came across that. Good to know. 😊 That said, as pointed out in another comment, we shouldn't use regex. It's slow and a memory hog. Even `String.Split` is faster like we [currently do in Key Vault](https://github.com/Azure/azure-sdk-for-net/blob/797a12c9b47c633d7d235089fe422b273f546e4f/sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs#L230-L236) (and I've been wanting to replace that, but never been a priority).